### DISCUSSION TITLE
ESM imports with assert are excluded from the build

### DISCUSSION THREAD

**Discussion by lizkenyon** [vercel/community#6758] [Help]
### Summary

Hi folks 👋 

We recently updated [our library](https://www.npmjs.com/package/@shopify/shopify-app-remix) to include ESM build. 
We use one case of import assertions in our typescript. 
(`import englishI18n from '@shopify/polaris/locales/en.json' assert {type: 'json'};`)

It appears that when this is built by Vercel, this file is not included. We can see that by running `vercel build` and looking in the `.vercel/output/functions/app.func/.vc-config.json` file. This file missing causing the application to crash.

Previously the CJS build worked, it used require instead of import, and with that the file was present in the final build.

This is affecting [all of our users ](https://github.com/Shopify/shopify-app-template-remix/issues/689)of our package that are currently using Vercel to deploy their applications. 

Is it possible to use import assertions with Vercel? Are there any recommended work arounds?

<img width="1658" alt="image" src="https://github.com/vercel/community/assets/23265671/69b06896-ec80-44f5-ada6-9e1b5e94d7c6">

### Example

https://github.com/lizkenyon/vercel-shopify-app-april

### Steps to Reproduce

1. Create a new application with the Shopify App Remix template
```
npm install -g @shopify/cli
shopify app init
```

2. Run `vercel build`
See that `import englishI18n from '@shopify/polaris/locales/en.json' assert {type: 'json'};` is not included in the `.vercel/output/functions/app.func/.vc-config.json` file. This causes the deployment to crash.

**Comment by jeffsee55** [comment]
I believe the fix for this has been released in the latest version. PR for the fix was [here](https://github.com/vercel/vercel/pull/11580)

**Comment by onsclom** [comment]
@jeffsee55 is right. Tested locally using `vc build` with v34.2 on that example, and it's fixed now!

Deploys on Vercel are still using CLI v34.1, but that should be updated later today.
